azure-pipelines.yml

```diff
@@ -1,56 +1,126 @@
+    - repository: templates
+      type: git
+      name: Enigmatry - Azure Pipelines Templates/enigmatry-azure-pipelines-templates
 
-- task: DotNetCoreCLI@2
-  displayName: 'Run Unit tests'
-  inputs:
-    command: 'test'
-    projects: '$(ProjectNamePrefix)**/*.Tests.csproj'
-    arguments: '--filter TestCategory=unit|TestCategory=smoke /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura /p:CoverletOutput=$(Build.SourcesDirectory)/TestResults/Coverage/'
-    configuration: $(buildConfiguration)
-    buildProperties: 
+stages:
+- stage: 'ci_build'
+  displayName: 'Build the packages'
+  jobs: 
+  - job: build_prerequisites
+    displayName: Get the version for the build
+    steps:
+    - checkout: self
+      fetchDepth: 0
+    - task: DotNetCoreCLI@2
+      displayName: Add Manifest for tool installation
+      inputs:
+        command: 'custom'
+        custom: 'new '
+        arguments: tool-manifest
+    - task: DotNetCoreCLI@2
+      displayName: Install minver
+      inputs:
+        command: 'custom'
+        custom: 'tool '
+        arguments: install minver-cli
+    - task: PowerShell@2
+      displayName: 'Set MinVer Version'
+      inputs:
+        targetType: inline
+        script: |
+          $version = dotnet minver -p preview
+          echo "##vso[build.updatebuildnumber]$version"
+          Write-Host "Setting MinVer to '$version'"
 
-- task: PublishCodeCoverageResults@1
-  displayName: 'Publish code coverage report'
-  inputs:
-    codeCoverageTool: 'Cobertura'
-    summaryFileLocation: '$(Build.SourcesDirectory)/**/coverage.cobertura.xml'
+  - job: 'Build_Package'
+    displayName: Build Package
+    dependsOn: build_prerequisites
+    steps:
+    - task: NuGetToolInstaller@1
+      displayName: Install NuGet Tool
+    - task: Cache@2
+      displayName: 'NuGet Cache'
+      inputs:
+        key: 'nuget | "$(Agent.OS)" | **/packages.lock.json,!**/bin/**,!**/obj/**'
+        restoreKeys: |
+            nuget | "$(Agent.OS)"
+            nuget
+        path: $(NUGET_PACKAGES)
+        cacheHitVar: 'NUGET_CACHE_RESTORED'
+    - task: NuGetCommand@2
+      displayName: Nuget Restore
+      inputs:
+        restoreSolution: '$(projectNamePrefix).sln'
+    - task: DotNetCoreCLI@2
+      displayName: 'Build package'
+      inputs:
+        command: 'build'
+        projects: '$(ProjectNamePrefix).sln'
+        configuration: $(buildConfiguration)
+        arguments: /p:TreatWarningsAsErrors=true
+    - task: DotNetCoreCLI@2
+      displayName: Run Unit Tests
+      inputs:
+        command: 'test'
+        projects: $(projectNamePrefix)**/*.Tests.csproj
+        arguments: '--filter TestCategory=unit|TestCategory=smoke --configuration $(buildConfiguration)'
+        nobuild: true
+    - task: DotNetCoreCLI@2
+      displayName: Pack the NuGet packages
+      inputs:
+        command: 'pack'
+        packagesToPack: '**/*.csproj'
+        configuration: '$(buildConfiguration)'
+        versioningScheme: 'byEnvVar'
```

Comment:
versioningScheme set to byBuildNumber doesn't work because you modified the build number above?

Reply:
byBuildNumber just returns A.B.C.